File: src/test/console/python/tests/iOSAppTest.py
from appium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.remote.remote_connection import RemoteConnection
import unittest
import time
import sys
sys.path.append('..')
import configs
import AutomationUtils

class iOSAppTest(unittest.TestCase):

  def setUp(self):
    device = AutomationUtils.getOnlineDevice('iOS')
    print('setUp - device:', device)
    self.driver = webdriver.Remote(AutomationUtils.kobitonServerUrl(), AutomationUtils.desiredCapabilitiesiOSApp(device))
    kobitonSessionId = self.driver.desired_capabilities.get('kobitonSessionId')
    print('https://portal.kobiton.com/sessions/{}'.format(kobitonSessionId))

  def tearDown(self):
    self.driver.quit()

  def test_ios_app(self):
    self.search_IFixit_on_home_screen()

  def search_IFixit_on_home_screen(self):
    print ('should allow to search iFixit on Home screen')
    self.driver.launch_app()
    time.sleep(2)
    self.driver.find_element_by_xpath("//XCUIElementTypeButton[@name='START A REPAIR']").click()
    time.sleep(2)
    self.driver.find_element_by_xpath("//XCUIElementTypeSearchField[@name='Search']").send_keys('Macbook Pro 2015')

    time.sleep(2)

    firstListResult = self.driver.find_elements_by_xpath("//XCUIElementTypeStaticText[contains(@label,'MacBook Pro')]")

    self.assertTrue(len(firstListResult) >= 33, 'The expected results are greater or equal to 33 results.')

if __name__ == '__main__':
  suite = unittest.TestLoader().loadTestsFromTestCase(iOSAppTest)
  isSuccess = unittest.TextTestRunner(verbosity=2).run(suite).wasSuccessful()
  sys.exit(not isSuccess)

File: ruby_rspec/iOSAppTest.rb
require "appium_lib"
require_relative "configs" 
require "selenium-webdriver"

describe "iOS App Test" do

    before(:all) do
        caps = DesiredCapabilities.new
        @driver = Appium::Driver.new({:caps => caps.desired_caps_ios_app[:caps],
        :appium_lib =>{ :server_url => caps.desired_caps_ios_app[:url]}},true).start_driver()
        @driver.manage.timeouts.implicit_wait = 300
    end

    after(:all) do
        @driver.quit
    end 

    it 'Navigate to some devices on Acura Support Community' do
        puts "Test case 1: it should allow to search some questions on Acura Support Community"
        @driver.find_element(:xpath, "//XCUIElementTypeButton[@name='START A REPAIR']").click
        sleep 10
        @driver.find_element(:xpath, "//*[@name='Car and Truck']").click
        sleep 10
        @driver.find_element(:xpath, "//*[@name='Acura']").click

        acura_text = @driver.find_element(:xpath, "//XCUIElementTypeNavigationBar").attribute("name")

        has_acura_integra = @driver.find_element(:xpath, "//XCUIElementTypeStaticText[@name='Acura Integra']").displayed?
        has_acura_mdx = @driver.find_element(:xpath, "//XCUIElementTypeStaticText[@name='Acura MDX']").displayed?
        has_acura_rl = @driver.find_element(:xpath, "//XCUIElementTypeStaticText[@name='Acura RL']").displayed?
        has_acura_tl = @driver.find_element(:xpath, "//XCUIElementTypeStaticText[@name='Acura TL']").displayed?
        has_acura_tsx = @driver.find_element(:xpath, "//XCUIElementTypeStaticText[@name='Acura TSX']").displayed?

        expect(acura_text).to eq('Acura')
        expect(has_acura_integra).to eq(true)
        expect(has_acura_mdx).to eq(true)
        expect(has_acura_rl).to eq(true)
        expect(has_acura_tl).to eq(true)
        expect(has_acura_tsx).to eq(true)
        @driver.close_app
    end

    it 'Search iFixit on Home screen' do
        puts "Test case 2: it should allow to search iFixit on Home screen"
        @driver.launch_app
        sleep 10
        @driver.find_element(:xpath, "//XCUIElementTypeButton[@name='START A REPAIR']").click
        sleep 10

        @driver.find_element(:xpath, "//XCUIElementTypeSearchField[@name='Search']").send_keys('Macbook Pro 2015')
        first_result = @driver.find_elements(:xpath, "//XCUIElementTypeStaticText[contains(@label,'MacBook Pro')]")
        @driver.find_element(:xpath, "//XCUIElementTypeButton[@name='Cancel']").click

        @driver.find_element(:xpath, "//*[@name='Search']").click
        @driver.find_element(:xpath, "//XCUIElementTypeSearchField[@name='Search']").send_keys('Acura')
        @driver.find_element(:xpath, "//XCUIElementTypeButton[@name='Categories']").click

        second_result = @driver.find_elements(:xpath, "//XCUIElementTypeStaticText[contains(@label,'Acura')]")

        expect(first_result.size).to be >= 39
        expect(second_result.size).to be >= 20
    end
end

